src/tools/gcc.lua

```diff
@@ -284,12 +284,20 @@
 -- Decorate include file search paths for the GCC command line.
 --
 
-	function gcc.getincludedirs(cfg, dirs, sysdirs)
+	function gcc.getincludedirs(cfg, dirs, sysdirs, frameworkdirs)
 		local result = {}
-		for _, dir in ipairs(dirs) do
+		for _, dir in ipairs(dirs or {}) do
 			dir = project.getrelative(cfg.project, dir)
 			table.insert(result, '-I' .. p.quoted(dir))
 		end
+
+		if (table.contains(os.getSystemTags(cfg.system), "darwin")
+			and cfg.frameworkdirs ~= nil and #cfg.frameworkdirs) then
```

Comment:
This should be on one line. This isn't using the right variable, shouldn't be using `cfg.` for `frameworkdirs`. Also, this is not consistent with the other `dirs` blocks by checking the value of `cfg.frameworkdirs`/`frameworkdirs`.

Reply:
Ok for the "one-line".

The use of cfg. is a copy-paste-is-evil mistake (from `gcc.getLibraryDirectories(cfg)`), that I previously fixed and re-introduced while rebasing for this PR...

The test on frameworkdirs value is made to minimize changes in code. Otherwise it will
require to add `, {}` on all `toolset.getincludedirs()` calls. I suggest to keep it for now.

Reply:
> The test on frameworkdirs value is made to minimize changes in code. Otherwise it will
require to add `, {}` on all `toolset.getincludedirs()` calls. I suggest to keep it for now.

Sorry to be pedantic, but isn't that why you have the `or {}` inside? Which is how the `sysdirs` loop works too. Checking if `frameworkdirs` is not `nil` isn't necessary because you're already doing `ipairs(frameworkdirs or {})`?